analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -222,7 +222,7 @@ class GoMod(
      * Return the list of all modules contained in the dependency tree with resolved versions and the 'replace'
      * directive applied.
      */
-    private fun getModuleInfos(projectDir: File): List<ModuleInfo> {
+    private fun getModuleInfos(projectDir: File): Map<String, ModuleInfo> {
```

Comment:
I remembered I tried to fix some bug with the replace directive but the approach ran into some other bug.
Maybe this is interresting to look at: https://github.com/oss-review-toolkit/ort/tree/go-mod-replace-directive-issue-fix

Reply:
I'll adapt the test like you did in that commit to at least verify this fix also in the test. Let's see if it works.

Reply:
I have added another replace directive to the test now, this showed that some more code changes were required to get the correct result, see the code comments.